frontend/src/components/forms/SelectInput.tsx

```diff
@@ -38,23 +44,34 @@ const SelectInput: FunctionComponent<Props> = ({
     },
   )
 
-  const registerOptions: RegisterOptions = {
-    required: otherProps.required,
-    valueAsNumber: castAsNumber,
+  if (register != null) {
+    const customOptions: RegisterOptions = _merge(
+      _pick(otherProps, ['required']),
+      registerOptions,
+    )
+
+    if (castAsNumber) {
+      customOptions.valueAsNumber = true
+    }
+
+    if (otherProps.required) {
+      customOptions.required = 'This field is required'
+      delete otherProps.required
+    }
```

Comment:
question (non-blocking): this seems a little confused? what's the relationship between `otherProps` and `customOptions`? why mutate the property list of `otherProps` with `delete`? I'm not concerned about performance but `delete` is generally not great for JS performance because the JIT can't assume the size of the object.